src/dune_rules/pkg_rules.ml

```diff
@@ -1089,7 +1083,8 @@ module Install_action = struct
       then None
       else (
         let dst = prepare_copy ~install_file ~target_dir entry in
-        hardlink_or_copy ~src ~dst;
+        (let src = Path.to_string src |> Unix.readlink |> Path.of_string in
```

Comment:
So the src is whatever the existing symlink was pointing to? How do we know if this is valid?

Reply:
if it's not valid, we're going to fail installing regardless.